### ISSUE TITLE
Feature Request - Download Results Into File

### ISSUE THREAD

**Issue by LordZeuss** [Ashfaaq18/OpenNetMeter#44]
This is a great program. One suggestion I have, would be to be able to possibly download results in a individual file, like a table. For example, I want to be able to click a "start recording" button of the sorts, monitor my traffic, and then "stop recording" and be able to have that result shown. I'll admit, I don't have a solid idea for how to display the results, but one possibility is to simply create a file, that shows the table results, that you can see in the detailed view tab.

I really like the reset button, to reset the active data, simply being able to download a results file would be awesome.

If I could essentially have a file that is the detailed view table, in a file, that would be perfect. 

I understand if this might not be possible, just something I think could be useful.

Great application! Love everything else about it!

**Comment by Ashfaaq18** [comment]
Hey @LordZeuss , thank you for the idea. Its an interesting one and i think i can make it work.

And btw, I'm currently implementing feature #39. In this, i plan on allowing the user to access their past 6 months data usage. 
Inside this 6 month range, the user can select either a specific day or multiple to view their usage. 
Is this somewhat related to what you want? 
I can implement the record and stop feature if the user wants more finer controls. For example, to retrieve the data usage within a range of minutes (e.g. start record at 17:40 and stop at 18:12).

>simply being able to download a results file would be awesome.

true, i plan on implementing an export feature. Haven't yet decided a file type to export as.

>I understand if this might not be possible, just something I think could be useful.

This shouldn't be too hard, I'll try it once i complete some of the other open issues.

>Great application! Love everything else about it!

Thank you!

**Comment by LordZeuss** [comment]
Absolutely! I think the 6 month range is a fantastic idea. That is the right concept for what I am looking for, and right now the "finer controls" method I use is to simply reset the collection, and monitor it that way. A start/stop would be more ideal, so if you're trying to gauge how much traffic a certain application uses, for certain tasks, you could at least stop the collection, so you can get a more accurate reading. If that makes any sense.

I'm right there with you as far as not knowing what file type to export as. I would imagine that it really depends on what kind of display you are going to do. From a table, or to simply list out the stats collected. I'm not sure if exporting the details to a "table" like view on a PDF, or if there is a way list the results in a HTML document. (I am not familiar with HTML so that may not be possible to generate on the fly.)

Even if there ends up not being a way to export, simply being able to do a start/stop and/or a rewind to see data usage from certain points in time, would be a addition to the program.

No rush on getting these features out, I think the program is very nice as it stands today and I will continue to use it. I do have one question, how does the program calculate the data usage? Does it measure the packets? Just curious from a technical standpoint.

**Comment by Ashfaaq18** [comment]
>No rush on getting these features out, I think the program is very nice as it stands today and I will continue to use it. 

Thanks again

>I do have one question, how does the program calculate the data usage? Does it measure the packets? Just curious from a technical standpoint.

I'm using a [Microsoft library](https://github.com/microsoft/perfview/blob/main/documentation/TraceEvent/TraceEventLibrary.md) which allows me to access a feature called [Event tracing for windows (ETW)](https://docs.microsoft.com/en-us/windows/win32/etw/about-event-tracing). Using this i can measure the data packets being transferred to and from the machine every second.
